iris_ugrid/tests/synthetic_data_generator/__init__.py

```diff
@@ -0,0 +1,115 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+"""
+Provides synthetic data generation capability in order to test ugrid loading and regridding.
+"""
+from pathlib import Path
+from string import Template
+import subprocess
+
+import netCDF4
+import numpy as np
+
+
+def create_file(
+    temp_file_dir,
+    dataset_name,
+    dataset_type,
+    n_faces=866,
+    n_times=1,
+    n_levels=39,
+):
+    """Create a synthetic NetCDF file with XIOS-like content.
+
+    Starts from a template CDL headers string, modifies to the input dimensions
+    then adds data of the correct size.
+
+    Parameters
+    ----------
+    temp_file_dir : str or pathlib.Path
+        The directory in which to place the created file.
+    dataset_name : str
+        The name for the NetCDF dataset and also the created file.
+    dataset_type : str
+        Name of the type of dataset to write to the file. The name = the name
+        of the template CDL string file to fetch from the `mesh_headers/`
+        directory.
+    n_faces, n_times, n_levels : int
+        Descriptors for the desired dimensions of the file's dataset.
+
+    Returns
+    -------
+    str
+        Path of the created NetCDF file.
+
+    """
+
+    # Fetch the specified CDL template type, modify the placeholders for the
+    # specified dimensions.
+    templates_dir = Path(__file__).parent / "mesh_headers"
+    template_filepath = templates_dir.joinpath(dataset_type).with_suffix(
+        ".txt"
+    )
+    with open(template_filepath) as file:
+        template_string = Template(file.read())
+    cdl = template_string.substitute(
+        DATASET_NAME=dataset_name,
```

Comment:
( According to suggestion above, would replace these with just `**kwargs` )

Reply:
https://github.com/SciTools-incubator/iris-ugrid/pull/38#discussion_r545952533